activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -396,6 +396,29 @@ def sanitize_limit(limit)
         end
       end
 
+      def prepare_keys_and_values_for_insert(inserts, table_name)
```

Comment:
Isn't this only called from InsertAll? If so, let's move it there.

Reply:
I left it in `database_statements.rb` because it calls to `with_yaml_fallback`, which is private. Do you want to make `with_yaml_fallback` public?

Reply:
Yeah, fine to make that public 👍

Reply:
(But with a nodoc comment so it isn’t public API)